tests/subsys/app_event_manager/src/modules/test_name_style_sorting.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#include "test_events.h"
+#include "name_style_events.h"
+
+#define MODULE test_name_style_sorting
+
+enum test_name_style_sorting_step {
+	TESTING_EVENT_NAME_STYLE,
+	TESTING_EVENT_NAME_STYLE_BREAK,
+	TESTING_EVENT_NAME_STYLE_END,
+};
+
+static enum test_name_style_sorting_step test_current_stage;
+static int test_name_style_cnt;
+
+
+/**
+ * @brief Assert that @a e event subscribers does not start inside @a er subscribers range
+ *
+ *
+ */
+#define app_zassert_subscriber_not_in(e, er, ...)                         \
+	zassert((_EVENT_ID(e)->subs_start < _EVENT_ID(er)->subs_start)  \
+		||                                                      \
+		(_EVENT_ID(e)->subs_start >= _EVENT_ID(er)->subs_stop), \
+		#e " subscriber overlaps " #er " subscriber", ##__VA_ARGS__)
+
+static void run_test_stage(enum test_name_style_sorting_step stage)
+{
+	test_current_stage = stage;
+
+	switch (stage) {
+	case TESTING_EVENT_NAME_STYLE:
+	{
+		struct event_name_style *et = new_event_name_style();
+
+		test_name_style_cnt = 0;
+		APP_EVENT_SUBMIT(et);
+		break;
+	}
+	case TESTING_EVENT_NAME_STYLE_BREAK:
+	{
+		struct event_name_style_break *et = new_event_name_style_break();
+
+		APP_EVENT_SUBMIT(et);
+		break;
+	}
+	case TESTING_EVENT_NAME_STYLE_END:
+	{
+		struct test_end_event *et = new_test_end_event();
+
+		et->test_id = TEST_NAME_STYLE_SORTING;
+		APP_EVENT_SUBMIT(et);
+		break;
+	}
+	default:
+		zassert_true(false, "Requested test stage is out of test scope");
+		break;
+	}
+}
+
+static bool app_event_handler(const struct app_event_header *aeh)
+{
+	if (is_test_start_event(aeh)) {
+		struct test_start_event *st = cast_test_start_event(aeh);
```

Comment:
Consider adding `const` - the received event is not modified here (also in other places)

Reply:
Done